### PR TITLE
Introduced GroovySample extension point

### PR DESCRIPTION
**Pull request by jglick**
Plan to introduce something in the `kubernetes` plugin. Also at some point the Declarative-specific samples should be moved out.

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/workflow/cps/samples/github-maven.groovy [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+   tools {
+      // Install the Maven version configured as "M3" and add it to the path.
+      maven "M3"
+   }
+
+   stages {
+      stage('Build') {
+         steps {
+            // Get some code from a GitHub repository
+            git 'https://github.com/jglick/simple-maven-project-with-tests.git'
```
Comment (olamy) [inline src/main/resources/org/jenkinsci/plugins/workflow/cps/samples/github-maven.groovy]: what about something from here https://github.com/cloudbeers?

File: src/main/resources/org/jenkinsci/plugins/workflow/cps/samples/scripted.groovy [hunk 2]
```diff
@@ -0,0 +1,25 @@
+node {
+   def mvnHome
+   stage('Preparation') { // for display purposes
+      // Get some code from a GitHub repository
+      git 'https://github.com/jglick/simple-maven-project-with-tests.git'
```
Comment (olamy) [inline src/main/resources/org/jenkinsci/plugins/workflow/cps/samples/scripted.groovy]: what about something from here https://github.com/cloudbeers?

File: src/main/resources/org/jenkinsci/plugins/workflow/cps/samples/github-maven.groovy (hunk 1 above)
Comment (jglick) [reply src/main/resources/org/jenkinsci/plugins/workflow/cps/samples/github-maven.groovy]: For this PR my goal was simply to introduce the extension point. Content changes can be done as follow-ups (and more easily reviewed as real `*.groovy` files).

File: src/main/resources/org/jenkinsci/plugins/workflow/cps/samples/scripted.groovy (hunk 2 above)
Comment (jglick) [reply src/main/resources/org/jenkinsci/plugins/workflow/cps/samples/scripted.groovy]: same

**Comment by jglick** [comment]
> Do you care about being able to suppress or filter default samples

Already done downstream in https://github.com/jenkinsci/kubernetes-plugin/pull/707.

**Pull request merged**